CMakeLists.txt

```diff
@@ -224,16 +224,25 @@ endif()
 
 add_subdirectory(portable)
 
-add_library(freertos_kernel STATIC
+list(APPEND FREERTOS_KERNEL_SOURCES
```

Comment:
Do not create yet another indirection of finding the list of files.

Use either the `add_library()` or `target_sources()` for the file lists.